code/modules/reagents/chemistry/reagents/impure_reagents/impure_medicine_reagents.dm

```diff
@@ -0,0 +1,412 @@
+	///The list of reagents we've affected
+	var/cached_reagent_list = list()
+	addiction_types = list(/datum/addiction/medicine = 1.75)
+
+/datum/reagent/inverse/healing/syriniver/on_mob_add(mob/living/living_mob)
+	if(!(iscarbon(living_mob)))
+		return ..()
+	var/mob/living/carbon/owner = living_mob
+	for(var/datum/reagent/reagent as anything in owner.reagents.reagent_list)
+		if(!(istype(reagent, /datum/reagent/medicine)))
+			continue
+		if(!(reagent.creation_purity > reagent.inverse_chem_val))//Only affect pure types
+			continue
+		reagent.creation_purity *= 0.8
+		cached_reagent_list += reagent
+	..()
+
+/datum/reagent/inverse/healing/syriniver/on_mob_delete(mob/living/living_mob)
+	if(!(iscarbon(living_mob)))
+		return ..()
+	if(!cached_reagent_list)
+		return ..()
+	for(var/datum/reagent/reagent as anything in cached_reagent_list)
+		if(!reagent)
+			continue
+		reagent.creation_purity *= 1.25
+	cached_reagent_list = null
+
+////Multiver
+//Inverse
+//Reaction product when between 0.2 and 0.35 purity.
+/datum/reagent/inverse/healing/monover
+	name = "Monover"
+	description = "A toxin treating reagent, that only is effective if it's the only reagent present in the patient."
+	ph = 0.5
+	addiction_types = list(/datum/addiction/medicine = 3.5)
+
+//Heals toxins if it's the only thing present - kinda the oposite of multiver! Maybe that's why it's inverse!
+/datum/reagent/inverse/healing/monover/on_mob_life(mob/living/carbon/owner, delta_time, times_fired)
+	if(owner.reagents.reagent_list > 1)
+		owner.adjustOrganLoss(ORGAN_SLOT_LUNGS, 0.5 * delta_time) //Hey! It's everyone's favourite drawback from multiver!
+		return ..()
+	owner.adjustToxLoss(-2 * REM * creation_purity * delta_time, 0)
+	..()
+	return TRUE
+
+///Can bring a corpse back to life temporarily (if heart is intact)
+///Makes wounds bleed more, if it brought someone back, they take additional brute and heart damage
+///They can't die during this, but if they're past crit then take increasing stamina damage
+///If they're past fullcrit, their movement is slowed by half
+///If they OD, their heart explodes (if they were brought back from the dead)
+/datum/reagent/inverse/penthrite
+	name = "Nooartrium"
+	description = "A reagent that is known to stimulate the heart in a dead patient, temporarily bringing back recently dead patients at great cost to their heart."
+	ph = 14
+	metabolization_rate = 1 * REM
+	addiction_types = list(/datum/addiction/medicine = 12)
+	overdose_threshold = 20
+	///If we brought someone back from the dead
+	var/back_from_the_dead
+
+/datum/reagent/inverse/penthrite/on_mob_dead(mob/living/carbon/owner)
+	var/obj/item/organ/heart/heart = owner.getorganslot(ORGAN_SLOT_HEART)
+	if(!heart || heart.organ_flags & ORGAN_FAILING)
+		return ..()
+	ADD_TRAIT(owner, TRAIT_STABLEHEART, type)
+	ADD_TRAIT(owner, TRAIT_NOHARDCRIT, type)
+	ADD_TRAIT(owner, TRAIT_NOSOFTCRIT, type)
+	ADD_TRAIT(owner, TRAIT_NOCRITDAMAGE, type)
```

Comment:
is trait_nodeath supposed to be in this list

Reply:
Yes! oops!